homeassistant/components/ozw/__init__.py

```diff
@@ -69,16 +75,61 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
     data_nodes = {}
     data_values = {}
     removed_nodes = []
+    manager_options = {"topic_prefix": f"{TOPIC_OPENZWAVE}/"}
 
-    @callback
-    def send_message(topic, payload):
-        mqtt.async_publish(hass, topic, json.dumps(payload))
+    if entry.unique_id is None:
+        hass.config_entries.async_update_entry(entry, unique_id=DOMAIN)
+
+    if entry.data.get(CONF_USE_ADDON):
+        # Do not use MQTT integration. Use own MQTT client.
+        host = entry.data.get(CONF_HOST)
+
+        if not host:
```

Comment:
If we can retrieve the discovery info, why wouldn't we always do that ? That way we know it's up to date